cmake/scripts/linux/ArchSetup.cmake

```diff
@@ -122,6 +122,22 @@ else()
   message(STATUS "memfd_create() not found")
 endif()
 
+include(CheckFunctionExists)
+set(CMAKE_REQUIRED_DEFINITIONS "-D_GNU_SOURCE")
+check_symbol_exists("STATX_BTIME" "linux/stat.h" HAVE_STATX)
+if(HAVE_STATX)
+  check_function_exists("statx" FOUND_STATX_FUNCTION)
+  if(FOUND_STATX_FUNCTION)
+    message(STATUS "statx is available")
+    list(APPEND ARCH_DEFINES "-DHAVE_STATX=1")
+  else()
+    message(STATUS "statx flags found but no linkable function : C library too old ?")
+  endif()
+else()
+  message(STATUS "statx() not found")
+endif()
+set(CMAKE_REQUIRED_DEFINITIONS "")
```

Comment:
@wsnipex OK to reset this to `""`?

Reply:
The previous sections do that after `check_symbol_exists` but I guessed it might be needed for `check_function_exists` as well which is why I reset it after the endif, rather than immediately.  Happy though to wait on confirmation that it's correct from @wsnipex .  I don't know that much about CMake files but have learnt a lot in the last few days.
Regarding the Android stuff, if @peak3d can confirm that `stat.h` is in the same location (`linux/stat.h`) then I can include the same check for that platform.

Reply:
yes, it's correct as is